bson/src/main/org/bson/internal/StringCodecHelper.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright 2008-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.bson.internal;
+
+import org.bson.BsonInvalidOperationException;
+import org.bson.BsonReader;
+
+import static java.lang.String.format;
+
+/**
+ * This class is not part of the public API. It may be removed or changed at any time.
+ */
+public final class StringCodecHelper {
+
+    private StringCodecHelper(){
+        //NOP
+    }
+
+    public static char decodeChar(final BsonReader reader) {
+        String string = reader.readString();
```

Comment:
Lets add a check if the bsonType isnt a string type. 

`throw new BsonInvalidOperationException(format("Invalid string type, found: %s", bsonType));`

Reply:
The method readString internally calls `verifyBSONType`, which already includes a type check and throws a `BsonInvalidOperationException` if the types don't match: [link to code](https://github.com/mongodb/mongo-java-driver/blob/bf03382f6b4c3b4e5f2c796564a6d1901f600a7b/bson/src/main/org/bson/AbstractBsonReader.java#L688). The current exception message is: '`X can only be called when CurrentBSONType is Y, not when CurrentBSONType is Z.`' 

Do we want to change this exception message with additional check?

Reply:
Adding the check would make it inline with the NumberCodecHelper.

Reply:
Thank you for clarification! I have added a check for BsonType.STRING.